### PR TITLE
feat: HTML to Markdown parser integration

### LINKED ISSUE #2847: Edit comment - Spaces are not displayed in edit box
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
Spaces should be visible in edit comment box

## Actual Result:
Spaces are not visible in edit comment box. 

## Action Performed:
1. Navigate to a conversation
2. Send a message with the following text 
```
Hello 
Space
Test
```
3. Right click on the message and click edit comment. 

## Workaround:
N/a

## Platform:
Where is this issue occurring?

 Web ✔️ 
 iOS
 Android
 Desktop App
 Mobile Web

**Version Number:** 1.0.43-0

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** 

image

**Expensify/Expensify Issue URL:** https://www.upwork.com/jobs/~01aab2d470107fe14c

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___
From @roryabraham https://expensify.slack.com/archives/C01GTK53T8Q/p1620849222131700

> When editing a comment, newlines do not appear in the text box.

### PR DESCRIPTION
**Pull request by pranshuchittora**
TODO merge this https://github.com/Expensify/expensify-common/pull/381 and then edit commit hash
### Details

A small step towards migration to markdown and deprecation of `text`

### Fixed Issues

Fixes #2847 

### QA Steps

1. Navigate to a conversation
2. Send a message with the following text
```
Hello 
Space
Test
```
3. Edit the comment.

The structure of the message should remain the same i.e. with new lines

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/32242596/123450804-baf39d00-d5fa-11eb-800d-325b79367b69.mp4

#### Mobile Web

https://user-images.githubusercontent.com/32242596/123450821-be872400-d5fa-11eb-9a74-920d02998c0e.mp4

#### Desktop

https://user-images.githubusercontent.com/32242596/123460217-ebd8cf80-d604-11eb-9fc8-d2dea6bbdcba.mov

#### iOS

https://user-images.githubusercontent.com/32242596/123459843-7240e180-d604-11eb-9cbc-f21ef6bb3eb6.mov

#### Android

https://user-images.githubusercontent.com/32242596/123453446-335b5d80-d5fd-11eb-9d67-7c27898367a5.mp4

### CODE REVIEW

**Comment by pranshuchittora** [comment]
https://user-images.githubusercontent.com/32242596/122986338-f2750600-d3bc-11eb-8f84-99b182a2de1b.mov

**Comment by pranshuchittora** [comment]
@roryabraham this PR is ready for review

File: src/pages/home/report/ReportActionContextMenu.js [hunk 1]
```diff
@@ -153,7 +154,8 @@ class ReportActionContextMenu extends React.Component {
      */
     getActionText() {
```
Comment (roryabraham) [inline src/pages/home/report/ReportActionContextMenu.js:155]: Let's change the method description here to:

```
Gets the markdown version of the message in an action.
```

**Review by roryabraham — COMMENTED** [review]
Thanks for this! Not much to review here, so looks good 👍 

However, before we can approve/merge, please update the description of this PR and include screen recordings for _all_ platforms.

**Review by roryabraham — CHANGES_REQUESTED** [review]
Actually, I think we should make this change in `ReportActionItemMessageEdit` instead of `ReportActionContextMenu`, because this doesn't work in the case where you edit the previous message using the `Up` arrow key:

1. Send a multiline message
1. Press the `Up` arrow key
1. Observe that the draft message in the `ReportActionItemMessageEdit` does not appear in multiple lines.

**Comment by pranshuchittora** [comment]
Hi @roryabraham I have updated the PR. Is this the correct approach, let me know for any changes.
Uploading Peek 2021-06-23 22-44.mp4…

**Review by roryabraham — CHANGES_REQUESTED** [review]
Not exactly correct – right now we're saving the draft message using the plain-text version of the message (i.e: `reportAction.message.text`). See the `getAction` function in `ReportActionContextMenu` [here](https://github.com/Expensify/Expensify.cash/blob/43bf806751d7643f8f9a005f19d0c77b73aba0fc/src/pages/home/report/ReportActionContextMenu.js#L122). Instead, I think we'll want to start passing the `reportAction.message.html` to `saveReportActionDraft`.

Once you've done that, the `draftMessage` prop of `ReportActionItemMessageEdit` will be html instead of text, and we shouldn't need the `reportAction` prop there. Then the only other change should be that we perform the HTML -> MD transform on the `draftMessage` prop of `ReportActionItemMessageEdit` in the constructor.

**Comment by pranshuchittora** [comment]
I have updated the PR

File: src/libs/reportUtils.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ function canEditReportAction(reportAction) {
     return reportAction.actorEmail === sessionEmail
         && reportAction.reportActionID
         && reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT
-        && !isReportMessageAttachment(lodashGet(reportAction, ['message', 0, 'text'], ''));
+        && !isReportMessageAttachment(lodashGet(reportAction, ['message', 0, 'html'], ''));
```
Comment (roryabraham) [inline src/libs/reportUtils.js:61]: Everything else looks good, but I don't think we need this change. When a reportAction is an image attachment, its `html` will contain an image tag. However, its `text` will just be `[Attachment]`.

**Review by roryabraham — CHANGES_REQUESTED** [review]

**Comment by pranshuchittora** [comment]
@roryabraham  I have updated all the required screen recordings

File: src/pages/home/report/ReportActionContextMenu.js (hunk 1 above)
Comment (roryabraham) [reply src/pages/home/report/ReportActionContextMenu.js:154]: Bump – this method doc is no longer accurate

**Review by roryabraham — CHANGES_REQUESTED** [review]
Sorry I missed this during my last review, but I still have the one comment about making the comment correct.

**Review by roryabraham — APPROVED** [review]
Thanks, LGTM 👍

**Comment by pranshuchittora** [comment]
@roryabraham can we merge this soon to avoid any merge conflicts.

**Comment by roryabraham** [comment]
Unfortunately the deploy comments are not working right now. This was deployed to staging yesteday.

**Pull request merged**